### PR TITLE
Feature/fix account errors

### PR DESCRIPTION
**Pull request by prachigarg19**
Resolves typescript error for `user/account` folder

### CODE REVIEW

File: src/features/user/Account/EditModal.tsx [hunk 1]
```diff
@@ -117,13 +123,13 @@ export const EditModal = ({
...
-        const res = await putAuthenticatedRequest(
+        const res: ModifyDonations = await putAuthenticatedRequest(
           `/app/subscriptions/${record?.id}?scope=modify`,
           bodyToSend,
           token,
           logoutUser
         );
-        if (res?.status === 'action_required') {
+        if (res?.status !== 'action_required') {
           window.open(res.response.confirmationUrl, '_blank');
```
Comment (Shreyaschorge) [inline src/features/user/Account/EditModal.tsx:133]: Gives following error
Property 'response' does not exist on type 'ModifyDonations'

File: src/features/user/Account/EditModal.tsx [hunk 2]
```diff
@@ -91,9 +91,15 @@ export const EditModal = ({
     setDisabled(false);
   }, [editModalOpen]);
 
+  interface bodyToSendType {
```
Comment (Shreyaschorge) [inline src/features/user/Account/EditModal.tsx:94]: In our codebase we follow PascalCase as a naming convention for all the interfaces

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/features/user/Account/EditModal.tsx (hunk 1 above)
Comment (prachigarg19) [reply src/features/user/Account/EditModal.tsx:133]: I'm not sure what response would be received for `status=action_required`. I'm waiting for information from Jorgo.

Comment (mohitb35) [reply src/features/user/Account/EditModal.tsx]: @prachigarg19 Please create an issue for this, and resolve later (once the scenario is replicable)

File: src/features/user/Account/components/AccountRecord.tsx
Comment (mohitb35) [inline src/features/user/Account/components/AccountRecord.tsx:1]: This commit (https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1765/commits/d77206ee43c53ccb62104a66ca25cf82769adaff#diff-6db4e901721bf323f56cc172cca4578d30b7f69f864f8e2d2f63d7e5e2529cb7) - may mean these changes are not needed.

**Review by mohitb35 — APPROVED** [review]

**Review by mohitb35 — APPROVED** [review]

**Pull request merged**